lineman/app/services/form_service.coffee

```diff
@@ -13,23 +13,49 @@ angular.module('loomioApp').factory 'FormService', ($rootScope, FlashService, $f
       422: 'unprocessableEntity'
       500: 'internalServerError'
 
+    prepare = (scope, model, options) ->
```

Comment:
I decided to split this out, because I want to keep all of the goodness of disabling the form, flash messages, broadcasting errors, etc. of submitting a form, but the submit function is a little different for upload.

This is slightly more complicated, but it still reads really well.